src/Simulation/Common/QuantumProcessorBase.cs

```diff
@@ -282,4 +283,18 @@ public virtual void OnMessage(string msg)
         }
 
     }
+
+    /// <summary>
+    /// A class that implements exception to be thrown when Operation is not supported by a QuantumProcessor.
+    /// </summary>
+    public class UnsupportedOperationException : NotImplementedException
```

Comment:
Why inherit from NotImplementedException? Do we expect code that catches NYI to catch these too? Also, could we use either of the built-in NotSupportedException/PlatformNotSuppotedException?